.openshift-ci/ci_tests.py

```diff
@@ -66,7 +67,15 @@ def run(self):
         print("Executing operator upgrade test")
         self.run_with_graceful_kill(
```

Comment:
If a test fails I would expect this to raise an error and the junit logs would not be gathered. What about using the `post_start_hook=set_dirs_after_start` convention used elsewhere in this file? Or just set `test_output_dirs` in a `OperatorE2eTest` `__init__()`?

Reply:
Hm, good point, this will only gather logs when we don't need them, :-)
Another way would be to change this artifact gathering into a context manager, or something?

Reply:
@gavin-stackrox do you suggest creating new `post_test` like `StoreTestResults`? Because `StoreArtifacts` doesn't add them to JUnit panel

Reply:
@ivan-degtiarenko actually, now that I think about it a bit more, maybe a better approach for the operator e2e tests is something similar to what is done with `UIE2eTest`. i.e. it is just a wrapper around `tests/e2e/run-ui-e2e.sh` which can then do all the error handling and JUnit artifact saving via `store_test_results()`.